pkg/aggregator/demultiplexer_serverless.go

```diff
@@ -147,11 +147,16 @@ func (d *ServerlessDemultiplexer) AddTimeSampleBatch(shard TimeSamplerID, sample
 	d.statsdWorker.samplesChan <- samples
 }
 
-// AddCheckSample doesn't do anything in the Serverless Agent implementation.
+// AddCheckSample is not supported in the Serverless Agent implementation.
 func (d *ServerlessDemultiplexer) AddCheckSample(sample metrics.MetricSample) {
 	panic("not implemented.")
 }
 
+// AddLateMetrics is not supported in the Serverless Agent implementation.
+func (d *ServerlessDemultiplexer) AddLateMetrics(samples metrics.MetricSampleBatch) {
+	panic("not implemented.")
```

Comment:
I agree, but this is a feature that they are interested to implement in the future hence it's presence this way for now.
I'll move the one above tho (`AddCheckSample` to the `DemultiplexerWithAggregator` to avoid having this panic L152, because indeed here it doesn't make sense).